src/main/java/cubicchunks/world/dependency/Dependency.java

```diff
@@ -26,31 +26,32 @@
 
 import java.util.Collection;
 
-import cubicchunks.world.ICubeCache;
 import cubicchunks.world.cube.Cube;
 import cubicchunks.worldgen.dependency.DependentCube;
 
-/*
- * TODO: Commenting
- */
+import javax.annotation.Nonnull;
+
 public interface Dependency {
 
 	/**
-	 * This collection specifies which cubes must be loaded for a given cube's requirements to be satisfied.
+	 * Provides a collection of requirements specifying the given cube's required cubes and the GeneratorStages they
+	 * need to be at.
 	 *
 	 * @return A collection of Requirements specifying the given cube's requirements.
 	 */
-	Collection<Requirement> getRequirements(Cube cube);
+	@Nonnull
```

Comment:
You can add `@MethodsReturnNonnullByDefault` and `@ParametersAreNonnullByDefault` annotations on the whole class to make methods `@Nonnull` by default, and then only add `@Nullable` when needed.